drfujibot.py

```diff
@@ -2940,43 +2961,77 @@ def processCommand(self, line, c, source_user, source_id=None, prefix=None):
 
                 output = ""
 
-                diff = now - last
-                if diff.days >= 1:
-                    more_coins = random.randint(0, 100)
+                timetoreset = datetime.timedelta(hours=self.daily_time) - datetime.timedelta(hours=last.hour, minutes=last.minute)
+                if timetoreset < 0:
+                    timetoreset = datetime.timedelta(hours=24) + timetoreset
+                #if diff.days >= 1:
+                if self.daily_type == "hours":
+                    if (diff.seconds/3600) >= self.daily_hours: 
```

Comment:
On this line, I would set a boolean value to say that coins should be awarded, and do the same in the "time" scenario. Then after both of those cases, add an "if" statement that decides whether to award coins based on the new boolean value. If true, then the code from more_coins onward will be executed to award coins. In other words, this avoids copy-pasting the same coin calculation code twice. Does that make sense?

Reply:
That makes a lot of sense. I knew copy-pasting chunks of code was definitely not the right solution, but I somehow didn't think to just use a boolean. Gonna make that change now!